testing/test_framework/src/firebase_test_framework.h

```diff
@@ -215,6 +215,27 @@ namespace firebase_test_framework {
     }                                                                       \
   }
 
+#define FLAKY_EXPECT_NULL(a)                                           \
+  {                                                                    \
+    auto a_result = (a);                                               \
+    if ((a_result) != nullptr) {                                       \
+      std::stringstream a_str;                                         \
+      a_str << a_result;                                               \
+      app_framework::LogError(                                         \
+          "Expected %s to be null, but it is not null. value(%s)", #a, \
+          a_str.str().c_str());                                        \
+      return false;                                                    \
+    }                                                                  \
+  }
+#define FLAKY_EXPECT_NOTNULL(a)                                              \
+  {                                                                          \
```

Comment:
technically this doesn't need a scope, but it looks nicer this way. :)

Reply:
Without the scope, the trailing ; will confuse the compiler - it ignores a ; after a {}.

(Although a lot of people like defining macros as "do { /* code_goes_here */ } while(0)" to give the semicolon something to do.)

Reply:
Ah right right.